src/main/java/com/user_service/services/UserAuthenticationService.java

```diff
@@ -45,23 +52,24 @@ public SignUpResponseDto signUpUser(RegisterUserDto registerUserDto) {
         try {
             SignUpRequest signUpRequest = SignUpRequest.builder()
                     .userAttributes(attrs)
-                    .username(registerUserDto.getEmail())
+                    .username(signUpUserDto.getEmail())
                     .clientId(CLIENT_ID)
-                    .password(registerUserDto.getPassword())
+                    .password(signUpUserDto.getPassword())
                     .build();
 
+            log.info("New user with email: {} created", signUpUserDto.getEmail());
             return new SignUpResponseDto(cognitoIdentityProviderClient.signUp(signUpRequest), null);
```

Comment:
Let's not use `null`, but `Optional.empty()` instead. This applies to other places as well.